### ISSUE TITLE
no-unused-vars with JSX

### ISSUE THREAD

**Issue by DaleJefferson** [eslint/eslint#2292]
*"eslint": "0.19.0"*

Since upgrading to eslint 0.19 i'm having false positive errors regarding unused vars and JSX. See below.

```
import {Col} from "react-bootstrap";

React.render(
    <Row>Row is used in the JSX</Row>,
    document.getElementById("divname")
);
```

```
error  Row is defined but never used                  no-unused-vars
```

**Comment by nzakas** [comment]
ESLint doesn't handle React semantics, so make sure you're using eslint-plugin-react, and if so, file a bug on that project.

**Comment by DaleJefferson** [comment]
Thanks for your quick response, once I added the react plugin this issue was fixed.
I am a little surprised that you say "ESLint doesn't handle React semantics" then you say in big letter on your site "The pluggable linting utility for JavaScript and **JSX**"

![screen shot 2015-04-12 at 07 37 42](https://cloud.githubusercontent.com/assets/375005/7104635/608de44c-e0e7-11e4-9925-e28feb1a8c80.png)

Since eslint 0.19 will break JSX support for a lot of people (not using the react plugin) I suggest maybe you make the react plugin more prominent on your website (I did not see it), include it or remove the JSX title.

**Comment by michaelficarra** [comment]
Wow, JSX === React now? Who knew...

**Comment by DaleJefferson** [comment]
@michaelficarra Your are correct I should not have to add eslint-plugin-react to have eslint work with JSX. This bug is still relevant and should be reopen.

**Comment by DaleJefferson** [comment]
@michaelficarra @nzakas Now with no React, just pure JSX. Issue still exists.

```
import Burndown from "./Burndown.jsx";

var burndownHtml = <Burndown>Burndown is used in the JSX</Burndown>;
```

```
src/Dashboard.jsx
  1:7  error  Burndown is defined but never used      no-unused-vars
```

**Comment by michaelficarra** [comment]
@DaleJefferson From the [JSX website](https://facebook.github.io/jsx/#transpilers):

> ### Transpilers
>
> These are a set of transpilers that all conform to the JSX syntax but use different semantics on the output:
>
> * [JSXDOM](https://github.com/vjeux/jsxdom): Create DOM elements using JSX.
> * [Mercury JSX](https://github.com/Raynos/mercury-jsx): Create virtual-dom VNodes or VText using JSX.
> * [React JSX](http://facebook.github.io/react/docs/jsx-in-depth.html): Create ReactElements using JSX.

Notice how the React semantics are explicitly distinguished from the JSX syntax. JSX is not React, nor does it imply React semantics. Do you understand now?

**Comment by DaleJefferson** [comment]
Thanks for taking the time to explain this. I found this pretty confusing I'm sure I won't be the only person. Maybe we can something to the FAQ?

**Comment by michaelficarra** [comment]
Maybe once that becomes frequent, it would be appropriate for the FAQ.

**Comment by nzakas** [comment]
We are working on making this clear. Bottom line: JSX is just syntax, React applies meaning on top of the syntax. Since ESLint is library agnostic, you need eslint-plugin-react to add the semantics.
